pkg/tfvars/openstack/openstack.go

```diff
@@ -62,3 +73,69 @@ func TFVars(masterConfig *v1alpha1.OpenstackProviderSpec, cloud string, external
 
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func getSwiftPublicURL(cloud string) (string, error) {
+	var swiftPublicURL string
+	serviceCatalog, err := getServiceCatalog(cloud)
+	if err != nil {
+		return "", err
+	}
+
+	for _, svc := range serviceCatalog.Entries {
+		if svc.Type == "object-store" {
+			for _, e := range svc.Endpoints {
+				if e.Interface == "public" {
+					swiftPublicURL = e.URL
+					break
+				}
+			}
+			break
+		}
+	}
+
+	if swiftPublicURL == "" {
+		return "", fmt.Errorf("cannot retrieve Swift URL from the service catalog")
+	}
+
+	return swiftPublicURL, nil
+}
+
+func getServiceCatalog(cloud string) (*tokens.ServiceCatalog, error) {
```

Comment:
would love to see some comments on the function regarding what it does.

Reply:
done